Hypesound

vertver opened the pull request:
Новый звук, с интеграцией Steam Audio и багфиксами

ForserX commented:
- [x] Новый звуковой движок (SDL + Phonon)
- [x] Обновлён SDL3 
- [x] Оптимизированы пайплайны

vertver committed:
Added steam audio disable define

v2v3v4 committed:
Fix GSC sounds files

vertver committed:
sound occlusion and snd_compression

ForserX committed:
Implement voice chat

ForserX committed:
Implement `HRTF` and `EAX` values change

vertver committed:
New sound system (SDL + SteamAudio)

Fix SDK and impl phonon reverb

SDL3 update (no nuget bitches)

ForserX committed:
Implement switch audio device

ForserX committed:
Попытка вправить HRTF

vertver committed:
Sound attenuation fix

vertver committed:
Target optimization stuff (smaa, taa and gtao now requires restart)

vertver committed:
SSLR perfomance fix

vertver committed:
attenuation fixes

vertver committed:
Reverb perfomance fix

ForserX committed:
Memory opt

vertver committed:
Accum fix

vertver committed:
Reverb optimization

Drombeys committed:
Fix rain steps volume

vertver committed:
hype fixes (leaks, attenuation, perf )

DimaKuzmin committed:
Cleaning Code* tbb MergeGeom

DimaKuzmin committed:
xrPhaseMergeLM * New Code

v2v3v4 committed:
Add profiler markers

ForserX committed:
Fix types and functions in `xrLevelBuilder`

Drombeys committed:
Fix `IterateInventory` script export

St4lker0k765 committed:
Add misfire mode switch

v2v3v4 committed:
Fix warnings

ForserX committed:
Improve thread switcher in builder

St4lker0k765 committed:
Add ability to disable 3rd person camera inertion

St4lker0k765 committed:
Fix some typos in `xrForms`

St4lker0k765 committed:
Remove `InitAutoFrameLineGroup` function

вместо неё используется InitAutoStaticGroup

St4lker0k765 committed:
Add frame window mode for buttons

St4lker0k765 committed:
Add `auto_framewindow` node support

DimaKuzmin committed:
xrPhase_MergeLM (выход из большой выборки)

ForserX committed:
Partial port `xrSound` on Linux

LostAlphaRus committed:
Фикс громкости звука, от источника к игроку (#143)

Co-authored-by: v2v3v4 <terran.erre@mail.ru>

Hrusteckiy committed:
Ability to add icon before answer

Using:
In `<phrase ` node add ```<icon_name ltx="1">wpn_pm</icon_name>``` or ```<icon_name>name_of_ui_texture</icon_name>```.
And then icon will be added before text. `ltx="1"` will read the icon from item's section, compatible with only 1-cell items by height. Without `ltx="1"` icon works properly when texture description has 1:1 ratio.

vertver committed:
Implement GPU markers

v2v3v4 committed:
Apply virtual cast methods

Hrusteckiy committed:
Track bar improvements

1) two new modes: token and bool, type `mode="token"` or `mode="flag"` to enable
2) ability to show track bar value on slider and under slider in token mode, need to add in node `out_num_of_signs` with needed accuracy
3) slider text color can be customized by adding node:
```
<slider_text_color>
	<t r="231" g="168" b="12"/>
	<d r="181" g="170" b="148"/>
	<e r="127" g="127" b="127"/>
	<h r="255" g="255" b="255"/>
</slider_text_color>```
or
```
<slider_text_color>
	<t r="181" g="170" b="148"/>
	<d color="gray"/>
	<e color="ui_1"/>
	<h color="red"/>
</slider_text_color>```

ForserX committed:
Restore `Stripify` geometry

v2v3v4 committed:
Remove memory stats

Hrusteckiy committed:
Additional static using for `CUI3tButton`

Also changed `vertical` param name to more explicit

Hrusteckiy committed:
Implement `UIArrowStepper`

Available modes: flag(bool); float, integer; token

Hrusteckiy committed:
Move backend UI configs into own folder

Hrusteckiy committed:
Rename `ui_arrow` to `UIArrow`

ForserX committed:
Improve pipelines

vertver committed:
Add SDL3 source code

ForserX committed:
Fix multiplayer launchability

vertver committed:
Apply new SDL3 to codebase

Co-authored-by: ForserX <sv3nk@yandex.ru>

ForserX committed:
Fix sound playback in editors

DimaKuzmin committed:
Вернул мой метод расположения LM в DDS файле (Уже поченил) (не делает засветы)

ForserX committed:
Remove `Loki` dependency

ForserX committed:
Add cmake option to disable discord

ViHtarb committed:
Скейлинг шрифтов для imgui оверлея в игре в зависимости от масштаба системы

ForserX committed:
Delete unused API

ForserX committed:
Fix `ImGui` input from keyboard after SDL3 update

DimaKuzmin committed:
tbb combinable

Drombeys committed:
Rework toolbar

Drombeys committed:
Improve name optick capture

GladiatorVS committed:
Rewritten texture picker in Weather Editor (test mode)

ForserX committed:
Make `There is no patrol path` assert as Release build

GladiatorVS committed:
Added button texture for language switching in the game menu

ForserX committed:
Fix wallmarks

ForserX committed:
Порядок небольшой

Shtrecker committed:
Fix new hud sound

Shtrecker committed:
Схлопнуть с `Аммо бонес визибле систем не финал`

Shtrecker committed:
Remove v2v3v4 detector stuff 😧

Shtrecker committed:
Add detector bolt throw animations

ForserX committed:
Hack for vanilla like engine

Drombeys committed:
Fix build

ForserX committed:
Fix working `remember` functions from Lua

ForserX committed:
Fix wndbar in editors

ForserX committed:
Remove duplicated `ImGui` api in editors

St4lker0k765 committed:
Fix missing textures in `CUIFrameWindow`

St4lker0k765 committed:
Fix messages duplication in MSVC Debugger

ForserX committed:
Remove duplicated theme code

Shtrecker committed:
Схлопнуть с `Add gunslinger mod type inertion`

St4lker0k765 committed:
Fix layered sounds

ForserX committed:
Fix mouse drawing after SDL3 update

ForserX committed:
Stable anim system

Drombeys committed:
Remove custom_spin file in gamedata_cs

St4lker0k765 committed:
Fix exit button in talk dialog for CS

St4lker0k765 committed:
Fix custom statics for `Clear Sky`

St4lker0k765 committed:
Add satiety indicator for `CS`

Co-Authored-By: Drombeys <23724295+Drombeys@users.noreply.github.com>

St4lker0k765 committed:
Add some features for future

St4lker0k765 committed:
Add second loadscreen progress bar for `CS`

wh1t3lord committed:
Fix when user saved game but they need to wait time_global from PREVIOUS tick so it means you need to wait the whole amount of time that you played in previous session, obviously it is a bug like I need to able to shoot after I loaded my save

St4lker0k765 committed:
Fix night vision postprocess

St4lker0k765 committed:
Fix stretched rank icon in character profile

St4lker0k765 committed:
Fix intro cutoff in `CS`

Drombeys committed:
Fix dosimeter showing

Co-authored-by: ForserX <sv3nk@yandex.ru>

St4lker0k765 committed:
Fix radiation progress shape in `CS`

St4lker0k765 committed:
Fix anomaly detection in `Clear Sky`

St4lker0k765 committed:
Remove renaming host client to `ServerAdmin`

St4lker0k765 committed:
Fix server list

St4lker0k765 committed:
Ignore missing `light_burn` hit sounds

St4lker0k765 committed:
Remove `xrRender_DS0` from list

St4lker0k765 committed:
Return map desc

St4lker0k765 committed:
Fix some issues with legacy UI

St4lker0k765 committed:
Fix MP UI in Clear Sky

St4lker0k765 committed:
Remove account check for public server

St4lker0k765 committed:
Add ability to customize shniaga speed

St4lker0k765 committed:
Fix map desc for CoP

St4lker0k765 committed:
Fix ammo in buy menu for `CS TDM`

St4lker0k765 committed:
Remove unnecessary platform checks

St4lker0k765 committed:
Replace GameSpy server with GSC ones

St4lker0k765 committed:
Fix inventory opening if dragging corpse of NPC

St4lker0k765 committed:
Remove CD-Key validation

ViHtarb committed:
Скейлинг размера шрифтов для сдк, попытки в скейлинг UITopBarForm

ViHtarb committed:
Исправлена проверка для "проводников", теперь он не сможет отвести вас туда, где находится.
Так же исправлена проверка для bind_signal_light

St4lker0k765 committed:
Add hack for buttons with single color (CS Moment)

St4lker0k765 committed:
Disable CD-Key by default

St4lker0k765 committed:
Fix `Capture the Artefact` game mode

St4lker0k765 committed:
Remove some platform checks from `UISkinSelector`

ViHtarb committed:
Исправлено отображение текста по вертикали на UI

St4lker0k765 committed:
Fix combobox arrow size on widescreen monitors

Co-authored-by: Viнt@rь <vihtarb@gmail.com>

wh1t3lord committed:
fix bloodsucker jump due to full disabling collision but we need to disable only dynamic colliding;

ViHtarb committed:
Чутка подвинул начало строчек в консоли, а то выглядит не очень)

ViHtarb committed:
UIHint теперь подстраивается по высоте по размеру текста внутри

Drombeys committed:
Fix state_mgr_direction scripts

St4lker0k765 committed:
Fix faction relationship progress bars

St4lker0k765 committed:
Fix missing texture warnings

St4lker0k765 committed:
Fix artefacts moving to belt by default

Drombeys committed:
Add resources for X-Ray 1.5.00

Drombeys committed:
Add resources for X-Ray 1.5.03

St4lker0k765 committed:
Make proper round map check

Drombeys committed:
Add resources for X-Ray 1.5.04

Drombeys committed:
Add resources for X-Ray 1.5.05

Drombeys committed:
Add resources for X-Ray 1.5.07

Drombeys committed:
Add resources for X-Ray 1.5.08

Drombeys committed:
Add resources for X-Ray 1.5.09

Drombeys committed:
Add resources for X-Ray 1.5.10

Drombeys committed:
Normalize encoding of scripts

K0cTep committed:
Fix escape treasure issuing from quest NPC

K0cTep committed:
Fix sudden death of quest NPC on escape level

Fixed sudden death of Drifter and Wolf brother

K0cTep committed:
Fix some treasures on escape level

Fixed treauses from army community

K0cTep committed:
Add treasure dialog for `esc_military_secret_trader`

K0cTep committed:
Fix dialog phrase for freedom fixer

K0cTep committed:
Fix freedom exo outfit upgrade

OSWALD-bit committed:
Add satiety indicator

Drombeys committed:
Add satiety indicator displaying

Drombeys committed:
Add satiety effect on health

Hozar2002 committed:
Fix dialog checks for weapons

Co-authored-by: Drombeys <dima.zolotov02@gmail.com>

Drombeys committed:
Add `ui_console` textures in `gamedata_cs` folder

Drombeys committed:
Add dialog menu selection by number

Hrusteckiy committed:
Add `TrashSlot` in Clear Sky

Hrusteckiy committed:
Fix bleeding icon on widescreen resolutions in Clear Sky

Hrusteckiy committed:
Fix inventory upgrade view in Clear Sky

Drombeys committed:
Remove temporary icons for quests in Clear Sky

Hozar2002 committed:
Fix texture offset in dialog window in Clear Sky

Hrusteckiy committed:
Fix upgrade icons on widescreen resolutions in Clear Sky

Hrusteckiy committed:
Add progress bars for items in Clear Sky

Hrusteckiy committed:
Apply `virtual_cells` for slots in Clear Sky

Hrusteckiy committed:
Fix big artefact icons on belt

Hrusteckiy committed:
Fix talk window on all resolutions

Hrusteckiy committed:
Fix community name in talk window

ForserX committed:
Fix for legacy AI

Hrusteckiy committed:
Fix offset for answer numbers

Hrusteckiy committed:
Fix text line alignment

Hrusteckiy committed:
Add `static_no_trade` support in Clear Sky

Hrusteckiy committed:
Partial backport UIBoosterInfo in Clear Sky

K0cTep committed:
Fix availability of traders money

Hrusteckiy committed:
Backport discounts feature for trade in Clear Sky

Hrusteckiy committed:
Fix non-tradable items in Clear Sky

By default it has 1% for saving vanilla behaviour.
It also backports CoP stuff

K0cTep committed:
Fix Sakharov money

K0cTep committed:
Make bandits in ambush invulnerable

K0cTep committed:
Fix looped award giving of Vintorez and artefact in Red Forest

K0cTep committed:
Fix issuing tasks on early jumping into anomaly

K0cTep committed:
Fix sound reload of `wpn_pkm` weapon in Clear Sky

Drombeys committed:
Fix sound reload of `wpn_svd` weapon in Clear Sky

Drombeys committed:
Add snd_bore parameter in weapons

Fixed sound bug when sounds would not play while stationary

Drombeys committed:
Fix sound reload of `wpn_spas12` and `wpn_wincheaster1300` weapons

K0cTep committed:
Fix treasure for Duty leader in Clear Sky

K0cTep committed:
Fix 6 Yantar treasures from 5 stalkers from Agroprom

Drombeys committed:
Fix `encyclopedia` closing tag

K0cTep committed:
Add fight ignoring for quest bandits

K0cTep committed:
Fix sound config for all killers

K0cTep committed:
Fix sound config of Wolf

K0cTep committed:
Add meet and bye phrases for stalker leader

K0cTep committed:
Fix additional tasks for digger and bandit leaders at garbage

Drombeys committed:
Backport postprocess system in Clear Sky

Shtrecker committed:
Backport knife in Clear Sky

Shtrecker committed:
Backport scope configs in Clear Sky

TODO: Locale, custom icons

K0cTep committed:
Fix missing active quest if player left dialogue with scientist

K0cTep committed:
Fix fifth phrase duty commander in `racya_dialog`

K0cTep committed:
Remove ability to enter dark valley before completing story missions

K0cTep committed:
Fix Vasyan icon in message if search his corpse

K0cTep committed:
Fix Leshiy phrase about first death killer after completing task with bridge

K0cTep committed:
Fix `agr_base_leader_sergant` playing sound

K0cTep committed:
Add 2 unused phrases for Saharov

Hrusteckiy committed:
Add `hud_message` custom static in Clear Sky

Hrusteckiy committed:
Fix non-tradable items

Hrusteckiy committed:
Add `cell_item_custom_text` custom static

Hrusteckiy committed:
Add `cell_item_custom_mark` custom static

Drombeys committed:
Fix warnings on infoportions registration

Drombeys committed:
Fix `add_exclusive_job` function in Clear Sky

Drombeys committed:
Delete legacy shaders in Clear Sky

Drombeys committed:
Add new fonts, shaders and textures in Clear Sky

Drombeys committed:
Move backend UI configs into own folder

Drombeys committed:
Apply changes to game resources in Clear Sky (without UI)

St4lker0k765 committed:
Replace `sdk_font_statistic` font in Editors

vertver committed:
крутые люди сделали дверь, я её запилил

St4lker0k765 committed:
Fix scope in CS (схлопнуть)

Drombeys committed:
Add scripts in CS (схлопнуть)

St4lker0k765 committed:
Backport load dialog in Clear Sky

St4lker0k765 committed:
Backport and apply new options in Clear Sky

St4lker0k765 committed:
Add ixray locale

St4lker0k765 committed:
Fix `inventory_new` xml files

Drombeys committed:
Add `game_loaded` event in Clear Sky

St4lker0k765 committed:
Adapt inventory in Clear Sky

Drombeys committed:
Add security checks to run Clear Sky

St4lker0k765 committed:
Add support for Clear Sky upgrades

St4lker0k765 committed:
Add `pda_spot` xml files

Drombeys committed:
Add `fire_queue_section` section in Clear Sky

St4lker0k765 committed:
Add security checks for voice chat

St4lker0k765 committed:
Remove `start_game_effect` (Схлопнуть)

St4lker0k765 committed:
Remove `kb_crouch_toggle` command

St4lker0k765 committed:
Fix `CUIOptionsItem` class for multiplayer

St4lker0k765 committed:
Backport `ui_detector_artefact` xml in Clear Sky

ForserX committed:
Restore sound env presets for `Shader Editor`

St4lker0k765 committed:
Fix `Clear Sky` crashes

St4lker0k765 committed:
Adapt basic HUD

St4lker0k765 committed:
Fix `car_panel` xml

St4lker0k765 committed:
Fix PDA UI

St4lker0k765 committed:
Add `ui_actor_sleep_screen` xml

St4lker0k765 committed:
Add `ui_ixray_spawn_icon` registration in Clear Sky

St4lker0k765 committed:
Backport load and save dialog in Clear Sky

St4lker0k765 committed:
Fix MP scripts in Clear Sky

St4lker0k765 committed:
Backport `pda` and `bind_stalker` scripts in Clear Sky

St4lker0k765 committed:
Fix UI stuff in Clear Sky

St4lker0k765 committed:
Fix grenades not stacking in inventory

ForserX committed:
Fix sound editor working without directories and access file

St4lker0k765 committed:
Add missing MP files

St4lker0k765 committed:
Fix `eat_portions_num` field in Clear Sky

St4lker0k765 committed:
Fix controller icons

St4lker0k765 committed:
Add missing locale in Clear Sky

St4lker0k765 committed:
Add eng locale in Clear Sky

St4lker0k765 committed:
Add working deathmatch for `Clear Sky`

St4lker0k765 committed:
Fix some UI issues

St4lker0k765 committed:
Fix map spots size in `Clear Sky`

St4lker0k765 committed:
Fix some UI issues

St4lker0k765 committed:
Add support for Clear Sky upgrades (Схлопнуть)

St4lker0k765 committed:
Fix some MP issues

St4lker0k765 committed:
Fix `highlight_item_slot` method

St4lker0k765 committed:
Fix journal and encyclopedia

Co-Authored-By: Hrusteckiy <47980896+Hrusteckiy@users.noreply.github.com>

ForserX committed:
Fix assert

ForserX committed:
Fix border window in editors

ForserX committed:
Fix build

ForserX committed:
Restore total size and total time in `Sound Editor`

ForserX committed:
Fix sound environiment render in SE

ForserX committed:
awda
